src/as-yamldata.c

```diff
@@ -1848,7 +1861,7 @@ as_yamldata_parse_distro_data (AsYAMLData *ydt, const gchar *data, GError **erro
 			gboolean header_found = FALSE;
 			GNode *root = g_node_new (g_strdup (""));
 
-			dep11_yaml_process_layer (&parser, root);
+			dep11_yaml_process_layer (&parser, root, error);
```

Comment:
Please also check for the error here and return NULL if an error is set (remember to issue a yaml_parser_delete (&parser); before that to not leak memory)